test/test_jit.py

```diff
@@ -3972,6 +3972,62 @@ def foo():
 
         self.assertEqual(foo(), [1, 2, 3, 4])
 
+    def test_mutable_list_pop_empty(self):
+        def test_pop_empty():
+            a = torch.jit.annotate(List[int], [])
+            return a.pop()
+
+        self.checkScriptRaisesRegex(test_pop_empty(), (), RuntimeError,
```

Comment:
To all the `checkScript` calls it turns things into TorchScript for you, so it needs the function itself instead of the return value (so this should be `self.checkScriptRaisesRegex(test_pop_empty, (), RuntimeError)`)